luatest/cbuilder.lua

```diff
@@ -0,0 +1,231 @@
+--
+-- @string path Option path.
+-- @param value Option value (int, string, table).
+local function cbuilder_set_replicaset_option(self, path, value)
+    assert(type(path) == 'string')
+    path = fun.chain({
+        'groups', self._group,
+        'replicasets', self._replicaset,
+    }, path:split('.')):totable()
+
+    -- <schema object>:set() validation is too tight. Workaround
+    -- it. Maybe we should reconsider this :set() behavior in a
+    -- future.
+    if value == nil then
+        local cur = self._config
+        for i = 1, #path - 1 do
+            -- Create missed fields.
+            local component = path[i]
+            if cur[component] == nil then
+                cur[component] = {}
+            end
+
+            cur = cur[component]
+        end
+        cur[path[#path]] = value
+        return self
+    end
+
+    cluster_config:set(self._config, path, value)
+    return self
+end
+
+-- Set an option of a particular instance in the selected replicaset.
+--
+-- @string instance_name Instance where the option will be saved.
+-- @string path Option path.
+-- @param value Option value (int, string, table).
+local function cbuilder_set_instance_option(self, instance_name, path, value)
+    assert(type(path) == 'string')
+    path = fun.chain({
+        'groups', self._group,
+        'replicasets', self._replicaset,
+        'instances', instance_name,
+    }, path:split('.')):totable()
+
+    cluster_config:set(self._config, path, value)
+    return self
+end
+
+-- Add an instance with the given options to the selected replicaset.
+--
+-- @string instance_name Instance where the config will be saved.
+-- @table iconfig Declarative config for the instance.
+local function cbuilder_add_instance(self, instance_name, iconfig)
+    local path = {
+        'groups', self._group,
+        'replicasets', self._replicaset,
+        'instances', instance_name,
+    }
+    cluster_config:set(self._config, path, iconfig)
+    return self
+end
+
+-- Return the resulting configuration.
+local function cbuilder_config(self)
+    return self._config
+end
+
+local cbuilder_mt = {
+    use_group = cbuilder_use_group,
+    use_replicaset = cbuilder_use_replicaset,
+    set_global_option = cbuilder_set_global_option,
+    set_group_option = cbuilder_set_group_option,
+    set_replicaset_option = cbuilder_set_replicaset_option,
+    set_instance_option = cbuilder_set_instance_option,
+    add_instance = cbuilder_add_instance,
+    config = cbuilder_config,
+}
+
+cbuilder_mt.__index = cbuilder_mt
+
+local function cbuilder_new(config)
+    config = table.deepcopy(config or base_config)
+    return setmetatable({
+        _config = config,
+        _group = 'group-001',
+        _replicaset = 'replicaset-001',
+    }, cbuilder_mt)
+end
+
+-- `cluster_config` module requires Tarantool 3.0.0 and newer
+-- Otherwise cbuilder will be an empty table.
+return has_cluster_config and { new = cbuilder_new } or {}
```

Comment:
This should be changed if you consider my comment above.

Reply:
Fixed.